Makefile

```
@@ -1,26 +1,35 @@
-default: build
+$(MAKE)FILES = $(shell find . -maxdepth 3 -type f -name Makefile)
+SUBDIRS   = $(filter-out ./,$(dir $($(MAKE)FILES)))
+ROOT_DIR = $(shell pwd)
 
-all: build test
+default: build
 
-test: build
-	cargo test
-	cargo test --features testutils
+all: test
 
 build:
-	cargo build --target wasm32-unknown-unknown --release -p soroban-token-contract
-	cargo build --target wasm32-unknown-unknown --release -p soroban-cross-contract-a-contract
-	cargo build --target wasm32-unknown-unknown --release -p soroban-atomic-swap-contract
-	cargo build --target wasm32-unknown-unknown --release
-	cd target/wasm32-unknown-unknown/release/ && \
-		for i in *.wasm ; do \
-			ls -l "$$i"; \
-		done
+	@for dir in $(SUBDIRS) ; do \
+		cd $$dir; \
+		$(MAKE) build || break; \
+		cd $(ROOT_DIR); \
```

You should be able to do this, which should mean you don't need the "ROOT_DIR" parameter.
```suggestion
		$(MAKE) -C $$dir build || break; \
```

---

Made the change in the new commit.